ui/v1/ui.js

```diff
@@ -304,6 +305,8 @@ class Ui extends PureComponent {
             def.domElement.appendChild(document.createTextNode(entry.hash));
             def.domElement.appendChild(element);
             console.log(`Marking ${entry.hash} in ${id}`);
+            // Let the node attach.
+            await sleep(0);
```

Comment:
How does this work? Is the point that sleep is non-blocking here so that thread is able to handle other calls on the stack?

Reply:
Ah, I thought I'd removed that.

The idea was to let the browser render the dom nodes -- the active code is in

      await new Promise((resolve, reject) => {
        requestAnimationFrame(resolve);
      });

in JsEditor

Reply:
Without this we get a zero offsetHeight on the widget element.